src/artemis/external_interaction/unit_tests/test_store_in_ispyb.py

```diff
@@ -69,22 +85,49 @@ def test_regex_string(dummy_ispyb, visit_path: str, expected_match: str):
 
 
 @patch("ispyb.open", new_callable=mock_open)
-def test_store_grid_scan(ispyb_conn, dummy_ispyb, dummy_params):
+def test_store_rotation_scan(
```

Comment:
Should: I think we're mocking out a bit too much here, which is why we're getting such poor coverage on `StoreRotationInIspyb`. Can we either in this test or in a new test mock just the upserts?

Reply:
I don't really understand. As fast as I can tell we are only mocking the connection (which is necessary at least for tests which run in CI) and nothing else? low coverage seems more down to not testing `_mutate_data_collection_params_for_experiment`

Reply:
I mean that later in this test we have lots of `when(x).y().thenReturn(z)`, which means we're never really testing any of the functionality of `x.y()`. I think it's fine to leave it for now though,given we have the additional tests on the mutate